src/bosh-director/spec/unit/api/task_remover_spec.rb

```diff
@@ -168,6 +171,46 @@ def tasks
           }.from(4).to(3)
         end
       end
+
+      context 'when there are tasks exceeding the retention period 1 day in the database' do
+        subject(:remover) do
+          TaskRemover.new(2000, 1, nil)
+        end
+
+        before do
+          make_n_tasks(1, checkpoint_time: inside_retention)
+          make_n_tasks(1, checkpoint_time: outside_retention)
+        end
+
+        it 'it removes the task outside retention' do
+          expect(remover).to_not receive(:remove_task).with(tasks[0])
+          expect(remover).to receive(:remove_task).with(tasks[1])
+
+          Timecop.freeze(Time.parse(inside_retention) + 60 * 60) do
+            remover.remove(default_type)
+          end
+        end
+      end
+
+      context 'when there is task exceeding the retention period 1 day in the database' do
+        subject(:remover) do
+            TaskRemover.new(2000, nil, [{ 'deployment' => 'deployment1', 'retention_period' => '1' }])
+        end
+
+        before do
+          make_n_tasks(1, checkpoint_time: outside_retention, deployment: 'deployment1')
+          make_n_tasks(1, checkpoint_time: outside_retention, deployment: 'deployment2')
+        end
+
+        it 'it removes the task outside retention' do
+          expect(remover).to receive(:remove_task).with(tasks[0])
+          expect(remover).to_not receive(:remove_task).with(tasks[1])
+
+          Timecop.freeze(Time.parse(inside_retention) + 60 * 60) do
+            remover.remove(default_type)
+          end
+        end
+      end
     end
```

Comment:
May be it also makes sense to have a test when retention period is set on general and deployment level?

Reply:
Done